src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java

```diff
@@ -0,0 +1,764 @@
+    }
+  }
+
+  @Override
+  public void setAcceleration(Double acceleration) {
+    this.acceleration = acceleration;
+  }
+
+  @Override
+  public void setAcceleration(Integer acceleration) {
+    this.acceleration = acceleration.doubleValue();
+  }
+
+  @Override
+  public void setAutoDisable(Boolean autoDisable) {
+    this.autoDisable = autoDisable;
+  }
+
+  @Override
+  public void setInverted(Boolean invert) {
+    mapper.setInverted(invert);
+  }
+
+  @Override
+  public void setMinMax(Double min, Double max) {
+    mapper.setMinMaxInput(min, max);
+  }
+
+  @Override
+  public void setMinMax(Integer min, Integer max) {
+    mapper.setMinMaxInput(min, max);
+  }
+
+  @Override
+  public void setPin(Integer pin) {
+    this.pin = pin + "";
+  }
+
+  @Override
+  public void setPin(String pin) {
+    this.pin = pin;
+  }
+
+  @Override
+  public void setRest(Double rest) {
+    this.rest = rest;
+  }
+
+  @Override
+  public void setRest(Integer rest) {
+    this.rest = rest.doubleValue();
+  }
+
+  @Override
+  public void setSpeed(Double degreesPerSecond) {
+    if (maxSpeed != -1 && degreesPerSecond > maxSpeed) {
+      speed = maxSpeed;
+      log.info("Trying to set speed to a value greater than max speed");
+    }
+    this.speed = degreesPerSecond;
+    for (String controller : controllers) {
+      ServiceInterface si = Runtime.getService(controller);
+      if (si.isLocal()) {
+        ((ServoController) Runtime.getService(controller)).servoSetVelocity(this);
+      } else {
+        send(controller, "servoSetVelocity", this);
+      }
+    }
+    broadcastState();
+  }
+
+  @Override
+  public void setSpeed(Integer degreesPerSecond) {
+    setSpeed(degreesPerSecond.doubleValue());
+  }
+
+  @Override
+  @Deprecated /* use setSpeed */
+  public void setVelocity(Double speed) {
+    setSpeed(speed);
+  }
+
+  @Override
+  @Deprecated /* use setSpeed */
+  public void setVelocity(Integer speed) {
+    setSpeed(speed);
+  }
+
+  // FIXME targetPos = pos, reportedSpeed, vs speed - set
+  @Override
+  public void stop() {
+    isSweeping = false;
+    sweeper = null;
+    for (String controller : controllers) {
+      ServiceInterface si = Runtime.getService(controller);
+      if (si.isLocal()) {
+        ((ServoController) Runtime.getService(controller)).servoSweepStop(this);
+      } else {
+        send(controller, "servoSweepStop", this);
+      }
+    }
+    broadcastState();
+  }
+
+  /**
+   * disable servo
+   */
+  public void stopService() {
+    super.stopService();
+    disable();
+  }
+
+  @Override
+  public void sync(ServoControl sc) {
+    if (sc == null) {
+      log.error("{}.sync(null)", getName());
+    }
+    syncedServos.add(sc.getName());
+  }
+
+  @Override
+  public void unsync(ServoControl sc) {
+    if (sc == null) {
+      log.error("{}.unsync(null)", getName());
+    }
+    syncedServos.remove(sc.getName());
+  }
+
+  @Override
+  public void waitTargetPos() {
+    // TODO Auto-generated method stub
```

Comment:
seems like this is a method that needs to block until a servo reaches the last requested position.